File: programming/lua/desktop.lua
#!/usr/bin/lua
lgi = require 'lgi'
gtk = lgi.Gtk
glib = lgi.GLib
gdk = lgi.Gdk
gobject = lgi.GObject
pixbuf = lgi.GdkPixbuf
gtk.init()

local window = gtk.Window {
   title = 'window', default_width = 400, default_height = 300,
   on_destroy = gtk.main_quit
}

if tonumber(gtk._version) >= 3 then
   window.has_resize_grip = true
end
-- -- create a gtk icon view and add icons to it
-- local icon_view = gtk.IconView
-- -- icon_view:set_column_spacing (10)
-- -- icon_view:set_row_spacing (10)
icon_view = gtk.IconView
{
   column_spacing = 10,
   row_spacing = 10
}

-- create store
store = gtk.ListStore (gdk.Pixbuf, gobject.TYPE_STRING)
icon_view:set_model (store) --icon_view:set_model ( model = node )

path = "/home/joshua/.config/awesome/icons/Faenza/apps/96/"
pix = pixbuf.Pixbuf.new_from_file ( path .. "libreoffice-writer.png" )

store:append(pix, "writer")

-- create a grid and add stuff to it.
local grid = gtk.Grid ()

grid:attach (icon_view, 0, 0, 1, 1)

window:add (grid)

-- show the window and start the loop
window:show_all()
gtk.main()


-- function MainWin.new()
--    local self = {}
--    setmetatable (self, MainWin)

--    self.w = gtk.window_new (gtk.Window_TOPLEVEL)
--    self.w:connect ('destroy', gtk.main_quit)
--    self.w:set_default_size (200, 250)
--    self.w:set_title ("Icon View Demo")

--    local sw = gtk.scrolled_window_new (nil, nil)
--    sw:set_policy (gtk.POLICY_AUTOMATIC, gtk.POLICY_AUTOMATIC)
--    self.w:add(sw)

--    self.icon_view = gtk.icon_view_new ()
--    sw.add(self.icon_view)

--    -- create store
--    self.store = gtk.list_store_new (3, glib.TYPE_INT, glib.TYPE_STRING,
-- 				    gdk.pixbuf_get_type())
--    self.icon_view:set_model (self.store)
--    self.icon_view:set_text_column (1)
--    self.icon_view:set_pixbuf_column (2)

--    -- insert some items
--    local iter = gtk.new "GtkTreeIter"
--    local pix

--    local names = { 'quit', 'open', 'redo', 'refresh', 'stop', 'save', 'save-as', 'select-color',
-- 		   'yes', 'no', 'zoom-fit' }

--    for i, name in ipairs(names) do
--       self.store:append(iter)
--       pix = self.icon_view:render_icon('gtk-' .. name, gtk.ICON_SIZE_DIALOG, "")
--       self.store:set(iter, 0, i, 1, name, 2, pix, -1)
--    end

--    self.w:show_all()
--    return self
-- end

File: programming/python/desktop.py
#!/usr/bin/python
from gi.repository import Gtk
class GridWindow (Gtk.Window):
    def __init__(self):
        Gtk.Window.__init__(self, title="Grid Example")
        path = "/home/joshua/.config/awesome/icons/Faenza/apps/96/"
        #create the gtk images and set the images they display
        image1 = Gtk.Image()
        image2 = Gtk.Image()
        image3 = Gtk.Image()
        image4 = Gtk.Image()
        image5 = Gtk.Image()
        image6 = Gtk.Image()
        image7 = Gtk.Image()
        image8 = Gtk.Image()
        image1.set_from_file(path + "libreoffice-writer.png")
        image2.set_from_file(path + "libreoffice-writer.png")
        image3.set_from_file(path + "libreoffice-writer.png")
        image4.set_from_file(path + "libreoffice-writer.png")
        image5.set_from_file(path + "libreoffice-writer.png")
        image6.set_from_file(path + "libreoffice-writer.png")
        image7.set_from_file(path + "libreoffice-writer.png")
        image8.set_from_file(path + "libreoffice-writer.png")

        #add the images to a button widget
        button1 = Gtk.Button(use_underline=False)
        button2 = Gtk.Button(use_underline=False)
        button3 = Gtk.Button(use_underline=False)
        button4 = Gtk.Button(use_underline=False)
        button5 = Gtk.Button(use_underline=False)
        button6 = Gtk.Button(use_underline=False)
        button7 = Gtk.Button(use_underline=False)
        button8 = Gtk.Button(use_underline=False)
        button1.add(image1)
        button2.add(image2)
        button3.add(image3)
        button4.add(image4)
        button5.add(image5)
        button6.add(image6)
        button7.add(image7)
        button8.add(image8)

        #connect the various signals
        # button1.connect("clicked", self.clicked1)

        #create the labels
        label1 = Gtk.Label ("label1")
        label2 = Gtk.Label ("label2")
        label3 = Gtk.Label ("label3")
        label4 = Gtk.Label ("label4")

        #add the buttons to the grid
        grid.add(button1)
        grid.attach(button2, 1, 0, 1, 1)
        grid.attach(button5, 2, 0, 1, 1)
        grid.attach(button6, 3, 0, 1, 1)
        grid.attach(button7, 4, 0, 1, 1)
        grid.attach(button8, 5, 0, 1, 1)
        grid.attach(label1, 0, 1, 1, 1)
        grid.attach(label2, 1, 1, 1, 1)
        grid.attach(button3, 0, 2, 1, 1)
        grid.attach(button4, 1, 2, 1, 1)
        grid.attach(label3, 0, 3, 1, 1)
        grid.attach(label4, 1, 3, 1, 1)

        #define what happens when you click a button
        # def clicked1 (self, button):
        #     print ("hello")



win = GridWindow()
win.connect("delete-event", Gtk.main_quit)
win.show_all()
Gtk.main()
